src/Components/SingleTodoItem.tsx

```diff
@@ -0,0 +1,56 @@
+import styled from 'styled-components';
+import useTodoContext from '../hooks/useTodoContext';
+import { ITodoItemProps } from '../Interfaces/interface';
+
+const SingleTodoItem = ({ item }: ITodoItemProps) => {
+  const { id, content, done } = item;
+  const { deleteTodo, doneItem, undoneItem } = useTodoContext();
+
+  const handleTodoItemClick = (id: number) => {
+    const handler = done ? undoneItem : doneItem;
+
+    handler(id);
```

Comment:
doneItem, undoneItem은 명사니까, doItem 혹은 changeItemDoneStatus 등의 동사적 이름으로 함수를 이름 지었으면 좋았을 것 같습니다. 

특히나 이 코드에서 handler의 타입이 불명확할뿐더러, undoneItem과 doneItem의 call signature가 달라질 경우 결국 분기해야 될 수도 있습니다. 따라서 이 경우는 다음과 같이 분리하는 것이 더 적합해 보입니다.

```suggestion
     if (done) {
        changeItemDoneStatus(id, false) 
     } else {
        changeItemDoneStatus(id, true)
     }
```
이렇게 작성하는 것이 좀 더 쉽겠죠?

혹은 `toggleItemDoneStatus(id)`라는 함수를 만들어서
```
    toggleItemDoneStatus(id)
```

아예 한줄 짜리 코드로 바꿔보시는 것도 좋을 듯 합니다.

Reply:
확실히 함수 이름짓는건 어렵습니다...